### ISSUE TITLE
No Crashes Reported after <NoLinkerSymbol> tag in .csproj

### ISSUE THREAD

**Issue by Arslan007** [microsoft/appcenter#2342]
When building Xamarin Form Apps with X64 with Native Tool Chain Support to get trusted build then build fails on RHBIND002 since RHBIND is 32bit so as per Microsoft suggested solution [https://github.com/microsoft/dotnet/blob/master/releases/UWP/net-native2.2/README.md#uwp-6212-february-9th-2021](https://github.com/microsoft/dotnet/blob/master/releases/UWP/net-native2.2/README.md#uwp-6212-february-9th-2021) We opted **`NoLinkerSymbol`** tag but after this build succeeds but there are no crashes are reported to AppCenter

**Comment by AnastasiaKubova** [comment]
Hi! Thanks for getting in touch with us! Could you please share the [verbose logs](https://docs.microsoft.com/en-us/appcenter/sdk/other-apis/xamarin#adjust-the-log-level)? Also could you please clarify do you build your app on the App Center portal or manually?

**Comment by Arslan007** [comment]
@AnastasiaKubova  thanks for your quick prompt response. We are using ADO to build the source code and its not done in AppCenter portal. I have enabled AppCenter Logging by this `AppCenter.LogLevel = LogLevel.Verbose;` but since this issue occurs in Release configuration so in our release build no App Center logs are captured in SeriLog. 
AppCenter only emits logs to Output window. I tried in debug Configuration as well but still AppCenter doesn't emit any `LogEvent` to `LoggerFactory`

Problem is its very hard to get Logs from `Release` build because it takes almost an hour to build in `Release Configuration` and then crash the app and restart the app and wait for another hour so you can see the second round of logs for sending crash.

Can you please confirm is there any easy way I can capture AppCenter logs in SeriLog/ or in file.

Thanks!

**Comment by AnastasiaKubova** [comment]
Unfortunately, the App Center DotNet SDK doesn't have any API to use a custom logger.

**Comment by Arslan007** [comment]
@AnastasiaKubova Thanks for getting back to this again.
I have attached the logs and I dont see any logs of sending crashes to AppCenter.
I can confirm I have test crash in my app via `Crashes.GenerateTestCrash();`but nothing is captured by SDK and not saved in `Microsoft.AppCenter.Storage` and thats why nothing is being sent.

[AppCenter Logs](https://github.com/microsoft/appcenter/files/7715376/Logs.txt)

Please let me know next step

**Comment by Arslan007** [comment]
Hi Again @AnastasiaKubova 

So more to this issue. I have sample Xamarin App which reproduces this issue.

[Sample App with Issue](https://github.com/microsoft/appcenter/files/7715743/UWPTest.zip)

Steps to Reproduce

1- Set you AppCenter Key in `App.xaml.cs`
2- Set the UWP as default Project (Should be already as default)
3- Choose the `X64` Release configuration and Run the UWP app with `X64 Release Configuration` 

<img width="781" alt="Screen Shot 2021-12-14 at 6 18 50 PM" src="https://user-images.githubusercontent.com/1511567/146105125-47c7e02a-c909-4aae-963e-6abad0112b30.png">

4- Hit the test crash button from main screen and verify your crash is being sent.

**P.S:** This solution is build with Visual Studio 2021 outside the AppCenter

**Comment by AnastasiaKubova** [comment]
Hello, @Arslan007 ! Thanks for the details and sample project. I checked it.

Based on logs that generate the app during local testing I found that after generating a crash with the `NoLinkerSymbol` property the app generates the `Fail Fast` exception.

Based on docs about [FastFail](https://docs.microsoft.com/en-us/dotnet/api/system.environment.failfast?view=net-5.0) there are no ways to catch this exception.

> This method terminates a process without running any active try/finally blocks or finalizers.

> The Environment.FailFast method writes the message string to the Windows Application event log, creates a dump of your application, and then terminates the current process. The message string is also included in error reporting to Microsoft.

This method calls hard termination of the process and it wouldn't call any events for handling exceptions.

The App Center SDK is an in-process crash reporter there are no ways to catch these crashes. We used a watson for catching crashes in UWP but it was dropped in the [2.2.1-preview](https://github.com/microsoft/appcenter-sdk-dotnet/blob/develop/CHANGELOG.md#uwp-5) version of App Center SDK due to restrictions and we have no plans to use it again.
